readthedocs/builds/tasks.py

```diff
@@ -453,3 +457,150 @@ def send_build_status(build_pk, commit, status, link_to_build=False):
             build.project.slug
         )
         return False
+
+
+@app.task(queue='web')
+def send_build_notifications(version_pk, build_pk, event):
+    version = Version.objects.get_object_or_log(pk=version_pk)
+
+    if not version or version.type == EXTERNAL:
+        return
+
+    build = Build.objects.get(pk=build_pk)
+
+    sender = NotificationSender(
+        version=version,
+        build=build,
+        event=event,
+    )
+    sender.send()
+
+class NotificationSender:
+
+    def __init__(self, version, build, event):
+        self.version = version
+        self.build = build
+        self.project = version.project
+        self.event = event
+
+    def send(self):
+        if self.event == WebHookEvent.BUILD_FAILED:
+            email_addresses = (
+                self.project.emailhook_notifications.all()
+                .values_list('email', flat=True)
+            )
+            for email in email_addresses:
+                try:
+                    self.send_email(email)
+                except Exception:
+                    log.exception(
+                        'Failed to send email notification. email=%s',
+                        email,
+                    )
+
+        old_webhooks = []
+        if self.event in [WebHookEvent.BUILD_FAILED, WebHookEvent.BUILD_PASSED]:
+            old_webhooks = (
+                self.project.webhook_notifications
+                .filter(events__isnull=True)
+            )
+
+        new_webhooks = (
+            self.project.webhook_notifications
+            .filter(events__name=self.event)
+        )
```

Comment:
Can't we replace this code with the following?

```python
self.project.webhook_notifications.filter(
 Q(events__isnull=True) | Q(events__name=self.event)
)
```

It doesn't seem we use the old_/new_ webhooks in a different way later in the code.

Reply:
> It doesn't seem we use the old_/new_ webhooks in a different way later in the code.

The old webhooks only listen to build failed and build passed events. But I'm planing on doing a data migration, so there isn't anything special about the old webhooks.